test/Tests/TestPartyCreationMenu.cpp

```diff
@@ -12,6 +12,37 @@ GAME_TEST(Items, GenerateItem) {
         pItemTable->GenerateItem(ITEM_TREASURE_LEVEL_6, 0, &item);
 }
 
+GAME_TEST(Issues, Issue198) {
+    // Check that items can't end up out of bounds of player's inventory.
+    test->playTraceFromTestData("issue_198.mm7", "issue_198.json");
+
+#if 0
+    // TODO(pskelton): This code crashes in glGetError, send help
+    Texture *image = assets->GetImage_ColorKey("item143");
+    (void) image->GetWidth();
+#endif
```

Comment:
Okay well the issue is GL is single thread only and by using texture we get a call to GL to load a texture from this test thread (this only applies to assets not previously loaded: if they were already initialized and drawn in the game thread then you should be fine to use texture in test thread).
By using an Image instead you can avoid any calls to GL but still get dimension details.

`Image* image = Image::Create(new ColorKey_LOD_Loader(pIcons_LOD, "item143", colorTable.TealMask.C16()));
int item143_width =  image->GetWidth();
image->Release();
`

Hope that makes sense / is what you're after?

Reply:
Ouch, yeah, makes perfect sense. Completely forgot that I'm running multiple threads here X)